### PR TITLE
Fix instances of old JuMP API

### PR DESCRIPTION
**Pull request by r-barnes**
I forgot some the first time around!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -498,10 +498,10 @@ When using this package via other packages such as [MathProgBase.jl](https://git
 using JuMP, Gurobi
 env = Gurobi.Env()
 
-m1 = Model(solver=GurobiSolver(env))
+m1 = Model(with_optimizer(Gurobi.Optimizer, env))
 ...
 
 # The solvers can have different options too
-m2 = Model(solver=GurobiSolver(env, OutputFlag=0))
+m2 = Model(with_optimizer(Gurobi.Optimizer, env))
```
Comment (odow) [inline README.md]: Missing `OutputFlag=0`

File: README.md [hunk 2]
```diff
@@ -327,7 +327,7 @@ natural algebraic approach.
 using JuMP, Gurobi
 
 # pass params as keyword arguments to GurobiSolver
-m = Model(solver=GurobiSolver(Presolve=0))
+m = Model(with_optimizer(Gurobi.Optimizer, Presolve=0))
```
Comment (blegat) [inline README.md]: `m` -> `model` according to JuMP style guide

Comment (r-barnes) [reply README.md]: Done.

File: README.md [hunk 3]
```diff
@@ -327,17 +327,17 @@ natural algebraic approach.
...
-@constraint(m, 30x + 33y <= 2100)
+@objective(model, Min, x + y)
+@constraint(model, 50x + 24y <= 2400)
+@constraint(model, 30x + 33y <= 2100)
 
-status = solve(m)
-println("Optimal objective: ",getobjectivevalue(m),
+status = solve(model)
+println("Optimal objective: ",getobjectivevalue(model),
 	". x = ", getvalue(x), " y = ", getvalue(y))
```
Comment (odow) [inline README.md]: `getobjectivevalue` -> `objective_value` and `getvalue` -> `value`.

File: README.md [hunk 4]
```diff
@@ -327,17 +327,17 @@ natural algebraic approach.
...
-@objective(m, Min, x + y)
-@constraint(m, 50x + 24y <= 2400)
-@constraint(m, 30x + 33y <= 2100)
+@objective(model, Min, x + y)
+@constraint(model, 50x + 24y <= 2400)
+@constraint(model, 30x + 33y <= 2100)
 
-status = solve(m)
-println("Optimal objective: ",getobjectivevalue(m),
+status = solve(model)
```
Comment (odow) [inline README.md]: `optimize!(model)` and no `status` is returned

File: README.md (hunk 3 above)
Comment (r-barnes) [reply README.md]: Fixed.

File: README.md (hunk 4 above)
Comment (r-barnes) [reply README.md]: Fixed.

File: README.md [hunk 5]
```diff
@@ -443,19 +443,19 @@ Note that you can use ``add_ivars!`` and ``add_bvars!`` to add multiple integer
 ```julia
 using JuMP, Gurobi
 
-m = Model(solver=GurobiSolver())
+model = Model(with_optimizer(Gurobi.Optimizer))
 
-@variables(m,begin
+@variables(model,begin
```
Comment (blegat) [inline README.md]: Missing space after comma

Comment (r-barnes) [reply README.md]: There was no space previously.

Comment (odow) [reply README.md]: These suggestions are all coming from the JuMP style guide. It's worth a read if you haven't already: https://pkg.julialang.org/docs/JuMP/DmXqY/0.19.0/style/#Whitespace-1

Comment (r-barnes) [reply README.md]: I'm objecting to the way a bug-fix pull request has been turned into an opportunity to update the existing code's style. I'm happy to do the bug fix and happy to cover things I've missed, but modifying parts of the code I didn't write to fit style conventions I don't know is not something I really want to do.

Comment (mlubin) [reply README.md]: @r-barnes This is understandably frustrating, but don't take these comments as a criticism of your changes. Our community has a convention (which is mentioned in the style guide) to fix style issues in code when it's touched, i.e., leaving the code tidier than when you found it. This is a sustainable way to propagate the style changes, reduce entropy in the codebase, and reduce the burden on maintainers. Your contributions are appreciated and will hopefully go more smoothly the next time around.

As an aside, this manual process will disappear once Julia has a proper linter that we can run as a CI pass.

Comment (r-barnes) [reply README.md]: Sounds good, thanks.

**Comment by odow** [comment]
Thanks @r-barnes

**Pull request merged**